ddp-workspace/projects/ddp-sdk/src/lib/services/serviceAgents/submissionManager.service.ts

```diff
@@ -122,13 +123,23 @@ export class SubmissionManager implements OnDestroy {
         const submitAnswerWithRetry: (submission: ActivityInstanceAnswerSubmission) => Observable<PatchAnswerResponse> =
             (submission) => {
                 return executeAnswerSubmission(submission).pipe(
+                    catchError(err => {
+                        if (err?.status === 422) {
+                            this.answerSubmissionFailureSubject.next(err);
+                            this.answerSubmissionPatch422ErrorSubject.next(this.getViolatedStableIds(err?.error?.violations));
+                            return EMPTY;
+                        } else {
```

Comment:
Consider creating a public observable, similar to answerSubmissionResponse$ that provides a stream of objects that represent the 422 body responses ($answerDataErrors: Observable<PatchDataError>  ?). Maybe you can come up with something better.
I think these sort of errors are different than the ones that we record in answerSubmissionFailureSubject. (Check,but probably OK that the 422 errors are not fed to answerSubmissionFailureSubject. Was looking and appears that all we ever do answerSubmissionFailureSubject emits is to direct users to error page, like an unrecoverable error.)

Then , in the activity, you can just subscribe to $answerDataErrors and do whatever you got to do with validation violations/don't have to worry about sorting between communication errors and validation errors.

Reply:
Fixed.